src/hammer-vlsi/synthesis/yosys/__init__.py

```diff
@@ -68,88 +119,256 @@ def tool_config_prefix(self) -> str:
+        self.append("exit")
+
+        # Create synthesis script.
+        syn_tcl_filename = os.path.join(self.run_dir, "syn.tcl")
+
+        with open(syn_tcl_filename, "w") as f:
+            f.write("\n".join(self.output))
+
+        # Build args.
+        args = [
+            self.get_setting("synthesis.yosys.yosys_bin"),
+            "-c", syn_tcl_filename,
+        ]
 
         if bool(self.get_setting("synthesis.yosys.generate_only")):
-            self.logger.info("Generate-only mode: command-line is " + \
-                             " ".join(args))
+            self.logger.info("Generate-only mode: command-line is " + " ".join(args))
         else:
-            # Temporarily disable colors/tag to make run output more readable
+            # Temporarily disable colours/tag to make run output more readable.
             # TODO: think of a more elegant way to do this?
             HammerVLSILogging.enable_colour = False
             HammerVLSILogging.enable_tag = False
-            self.run_executable([run_script]) # TODO: check for errors
+            self.run_executable(args, cwd=self.run_dir) # TODO: check for errors and deal with them
             HammerVLSILogging.enable_colour = True
             HammerVLSILogging.enable_tag = True
 
         return True
 
+    def write_sdc_file(self) -> bool:
+        with open(self.mapped_sdc_path,'w') as f:
+            # TODO: why buf cell and not inv??
+            # custom sdc constraints
+            f.write("set_driving_cell sky130_fd_sc_hd__buf_1\n" )
+            f.write("set_load 5\n")
+
+    def block_append(self,commands) -> bool:
+        for line in commands.split('\n'):
+            self.append(line.strip())
+            # if line.strip().startswith('#') or (line==""):
+            #     self.append(line.strip())
+            # else:
+            #     self.verbose_append(line.strip())
+        return True
+
     #========================================================================
     # synthesis main steps
     #========================================================================
     def init_environment(self) -> bool:
-        # TODO: currently using OpenROAD's default synthesis script
+
+        # set variables to match global variables in OpenLANE
+        self.driving_cell = "sky130_fd_sc_hd__inv_1" # SYNTH_DRIVING_CELL
+        self.driving_cell_pin = "Y" # SYNTH_DRIVING_CELL_PIN
+        self.buffer_cell = "sky130_fd_sc_hd__buf_2 A X"
+
+        time_unit = self.get_time_unit().value_prefix + self.get_time_unit().unit
+        clock_port = self.get_clock_ports()[0]
+        self.clock_port_name = clock_port.name
+        time_unit = "ps" # yosys requires time units in ps
+        self.clock_period = int(clock_port.period.value_in_units(time_unit))
+        self.clock_uncertainty = int(clock_port.period.value_in_units(time_unit))
+        self.clock_transition = 0.15 # SYNTH_CLOCK_TRANSITION
+
+        self.synth_cap_load = 33.5 # SYNTH_CAP_LOAD
```

Comment:
How is this chosen? It might be interesting to extract it from the driving cell...

Reply:
Unclear, but I agree. Just left the hard-coded value from the OpenROAD+sky130 flow in the interest of time.